composer/core/state.py

```diff
@@ -540,14 +546,87 @@ def state_dict(self) -> Dict[str, Any]:
 
         return state_dict
 
-    def load_model_state(self, state_dict: Dict[str, Any], strict: bool):
+    def _apply_required_algorithms(
+        self,
+        state: Dict[str, Any],
+        logger: Logger,
+        exclude_algorithms: Optional[List[str]] = None,
+    ):
+        """Applies required algorithms which haven't been specified and aren't in the exclude list.
+
+        Args:
+            state (Dict[str, Any]): State from checkpoint.
+            logger (Logger): Logger to use.
+            exclude_algorithms (List[str], optional): List of algorithm names to exclude.
+        """
+        import composer.algorithms as algorithms  # type: ignore imports used in `eval(representation)`
+
+        # Get repr of existing algorithms
+        state_algos = {}
+        for algo in self.algorithms:
+            if algo.required_on_load():
+                if type(algo) not in state_algos:
+                    state_algos[type(algo)] = []
+                state_algos[type(algo)].append(algo.__repr__())
+
+        missing_algorithms = []
+        for algo_name, serialized_value in state['algorithms']:
+            if hasattr(algorithms, algo_name) and getattr(algorithms, algo_name).required_on_load():
+                algo = eval(f"algorithms.{serialized_value['repr']}")
+                if exclude_algorithms is None or type(algo).__qualname__ not in exclude_algorithms:
+                    if type(algo) in state_algos and not serialized_value['repr'] in state_algos[type(algo)]:
```

Comment:
Could you explain when this warning would get hit?

Reply:
For example, I add `BlurPool` to replace convs and pooling. In a second run, I add BlurPool but only replace convs. It would then hit this warning